dvc/checkout.py

```diff
@@ -30,7 +30,7 @@ def _changed(path_info, fs, obj, cache):
         return True
 
     try:
-        actual = get_hash(path_info, fs, obj.hash_info.name)
+        actual = get_hash(path_info, fs, obj.hash_info.name, cache)
```

Comment:
`get_hash` shouldn't be used directly, we should use `stage()` instead. Again, this will be addressed a bit later, need to move staging/state/dir_info code around a bit first.